azure/src/main/scala/quasar/blobstore/azure/Azure.scala

```diff
@@ -16,93 +16,93 @@
   def mkStdStorageUrl(name: AccountName): StorageUrl =
     StorageUrl(s"https://${name.value}.blob.core.windows.net/")
 
-  def mkCredentials[F[_]: ConcurrentEffect](cred: Option[AzureCredentials]): F[Expires[ICredentials]] =
-    cred match {
-      case None =>
-        Expires.never(
-          new AnonymousCredentials: ICredentials).pure[F]
-
-      case Some(AzureCredentials.SharedKey(accountName, accountKey)) =>
-        Expires.never(
-          new SharedKeyCredentials(accountName.value, accountKey.value): ICredentials).pure[F]
-
-      case Some(AzureCredentials.ActiveDirectory(clientId, tenantId, clientSecret)) =>
-        rx.publisherToStream[F, AccessToken](
-          (new ClientSecretCredentialBuilder)
-            .clientId(clientId.value)
-            .tenantId(tenantId.value)
-            .clientSecret(clientSecret.value)
-            .tokenRefreshOffset(Duration.ofMinutes(60))
-            .build()
-            .getToken((new TokenRequestContext)
-              .setScopes(List("https://storage.azure.com/.default").asJava)))
-          .compile
-          .lastOrError
-          .map(tk => Expires(new TokenCredentials(tk.getToken), tk.getExpiresAt))
-    }
-
-  def mkContainerUrl[F[_]](cfg: Config)(implicit F: ConcurrentEffect[F]): F[Expires[ContainerURL]] =
-    F.catchNonFatal(new URL(cfg.storageUrl.value)) flatMap { url =>
-      // Azure SDK changed its logging behavior in 10.3.0 (compared to 10.2.0).
-      // In 10.2.0 the standard `new PipelineOptions` could be used, but in 10.3.0
-      // this results in messages being logged to console in eg quasar's repl
-      // (despite having no console logger configured).
-      // Still not having a full understanding of how this logging is supposed to work
-      // exactly (e.g. `withLogger`) but adding `LoggingOptions` with
-      // `disableDefaultLogging = true` seems to be fixing it.
-      // As an aside, in case we need to look at this in the future,
-      // Azure SDK has 2 loggers for some reason:
-      // - `com.microsoft.azure.storage.blob.LoggingFactory`
-      // - `Azure Storage Java SDK`
-
-      Functor[F].compose[Expires].map(mkCredentials(cfg.credentials)) { creds =>
-        val pipeline =
-          StorageURL.createPipeline(
-            creds, new PipelineOptions().withLoggingOptions(new LoggingOptions(3000, true)))
-
-        new ServiceURL(url, pipeline).createContainerURL(cfg.containerName.value)
+  def getAccessToken[F[_]: Async: ContextShift](ad: ActiveDirectory): F[Expires[AccessToken]] =
+    Sync[F].delay {
+      (new ClientSecretCredentialBuilder)
+        .clientId(ad.clientId.value)
+        .tenantId(ad.tenantId.value)
+        .clientSecret(ad.clientSecret.value)
+        .tokenRefreshOffset(Duration.ofMinutes(60))
+        .build()
+        .getToken((new TokenRequestContext)
+          .setScopes(List("https://storage.azure.com/.default").asJava))
+    } flatMap reactive.monoToAsync[F, AccessToken] map (t => Expires(t, t.getExpiresAt()))
```

Comment:
Ironically, this style is the exact opposite of what we normally do. Normally, `Sync[F] delay {` (no dot because curly brace following), and then on the final line we would normally split the result of `delay` out into an intermediate `val` and then use dot syntax for the `flatMap` and `map`.

Reply:
I am aware of the usual style at a high-level and I try to follow that, but in practise there are details coming up where the style works out less well (from my perspective any way) or it doesn't work out at all. 

In this example, for me the dot from `Sync[F].delay` is tighter than dot in `something.map` (and although it isn't, I treat it more like a dot in `mypackage.myfunction`)

The rule of "intermediate `val`s" I wasn't aware of. I was just fiddling my way towards something that was 1 expression, compiled and looked ok to me.

Another way to look at the result I came up with is that with a single change of adding parens around  `{ (new ClientSecretCredentialBuilder)....}` then it becomes fully compliant to the style again. So from that perspective I was just removing unnecessary parens.

BTW An example where the style doesn't work is the `setCredentials` function below where you will get the "postfix operator needs to be enabled" error.

Anyway, I think it would be great if we had another try introducing scalafmt. Should work out better now the old quasar mono/mega-repos has splintered off into multi-repos. And also scalafmt has evolved further.